### ISSUE TITLE
Guake does not open on display with mouse in gnome

### ISSUE THREAD

**Issue by xhpohanka** [Guake/guake#2115]
**Describe the bug**
Guake does not open on display with mouse in gnome, it is still opened on primary monitor regardless the settings. Due to wayland I'm using `guake-toogle` on global shortcut. I'm not sure if it is a wayland issue.

**Expected behavior**
When configured guake window should open on the display with mouse cursor.

-------------------------------------------------------------------------------

Please run `$ guake --support`, and paste the results here. Don't put backticks (`` ` ``) around it! The output already contains Markdown formatting. And make sure you run the command **OUTSIDE** the Guake.

$ guake --support
<details><summary>$ guake --support</summary>

Guake Version:		3.9.1.dev0

Vte Version:		0.68.0

Vte Runtime Version:	0.68.0

--------------------------------------------------
GTK+ Version:		3.24.34

GDK Backend:		<GdkX11.X11Display

--------------------------------------------------
Desktop Session: gnome

--------------------------------------------------
Display: :0

RGBA visual: True

Composited: True

* Monitor: 0 - XWAYLAND0
    * Geometry:		1920 x 1200 at 1920, 0
    * Size:		540 x 350 mm²
    * Primary:		True
    * Refresh rate:	59.884 Hz
    * Subpixel layout:	unknown
* Monitor: 1 - XWAYLAND1
    * Geometry:		1920 x 1080 at 0, 120
    * Size:		300 x 170 mm²
    * Primary:		False
    * Refresh rate:	59.962 Hz
    * Subpixel layout:	unknown

**Comment by xhpohanka** [comment]
More over I have noticed that when I connect external monitor to my notebook, which is then used as a primary, guake is still opening on internal one...

**Comment by xhpohanka** [comment]
Next observation when *display with mouse is selected*, guake is shown on the screen where last XWayland application was hovered. It's a bit difficult to follow when you have many applications :D

**Comment by nmay231** [comment]
Upgrading to the latest (3.9.1.dev12) seemed to work for me.

**Comment by xhpohanka** [comment]
Still not working in (3.9.1.dev12) for me
The #2121 is probably the same issue.

**Comment by nmay231** [comment]
Actually, it's also not working for me. Not sure why I thought it did :sweat_smile:

**Comment by Ivaylo-Korakov** [comment]
I am facing the same issue well on a brand-new Ubuntu 22.04 install while installing Guake through apt.
Here is the output of `guake --support`

<details>
  <summary>Expand `guake --support`</summary>
  
```
❯ guake --support
<details><summary>$ guake --support</summary>

Guake Version:		3.8.5

Vte Version:		0.68.0

Vte Runtime Version:	0.68.0

--------------------------------------------------
GTK+ Version:		3.24.33

GDK Backend:		<GdkX11.X11Display

--------------------------------------------------
Desktop Session: ubuntu

--------------------------------------------------
Display: :0

RGBA visual: True

Composited: True

* Monitor: 0 - XWAYLAND0
    * Geometry:		2560 x 1440 at 1920, 0
    * Size:		600 x 340 mm²
    * Primary:		True
    * Refresh rate:	143.913 Hz
    * Subpixel layout:	unknown
* Monitor: 1 - XWAYLAND1
    * Geometry:		1920 x 1080 at 0, 0
    * Size:		520 x 290 mm²
    * Primary:		False
    * Refresh rate:	59.962 Hz
    * Subpixel layout:	unknown
```
</details>


I wanted to note that there is indeed weird behavior. If you have no windows open on the second monitor even if you have your mouse there it will still open on the main one. If you have a window open on the second monitor of some kind then guake will appear on both screens depending on where the mouse is just like normal. 

TL;DR; 
Try to open an app/window on every monitor and it might work :smile:

**Comment by MattSom** [comment]
Fresh Ubuntu 22.04 install. Same issue.

**Comment by nmay231** [comment]
For what it's worth, you can use `Win`+click+drag can move it to the other screen. It also seems to fix it, at least temporarily.

**Comment by chalbersma** [comment]
I've found I have to use a X on Wayland window and guake will follow that window back and forth.

**Comment by alanmcg** [comment]
Solution for me was to set the guake-toggle keyboard shortcut in *gnome keyboard prefs* not just in quake prefs.

![image](https://github.com/Guake/guake/assets/43033315/bc0fac43-a6af-49a8-b492-c711eebbfcaf)

You can see i use the "section" key §, which I now have to copy paste from wikipedia :P

**Comment by nicolaslima321** [comment]
Still facing it with Ubuntu 25.10, does anyone know a definitive solution?

**Comment by maxpolzin** [comment]
Also on Ubuntu 26.04

**Comment by adam8kac** [comment]
This is a Wayland/XWayland issue with how Guake queries the pointer position -> on Wayland sessions Guake runs under XWayland (GDK Backend: X11Display in the support dumps above), and X11 pointer queries only get updated when the cursor is over an XWayland-backed window. That matches exactly what's reported here: it works while hovering XWayland apps and "sticks" to the last monitor seen once the cursor moves over a native Wayland surface. Here's a working solution for GNOME/Mutter-based Wayland sessions - a GNOME Shell extension that reads the pointer position directly from the compositor instead of relying on Guake's own X11-based detection, and moves the Guake window to the correct monitor whenever it's shown: https://github.com/adam8kac/guake-follow-cursor

Won't help on non-GNOME compositors since it's a shell extension, but it fully resolves the "sticks to last monitor" behavior on GNOME Wayland.

**Comment by anhtumai** [comment]
I agree with @adam8kac that finding the latest position of cursor on Gnome with XWayland backend is impossible for a standalone app at the moment, requires a Gnome Extension to do so.

Setting a GNome extension up is quiet complicated, I am leaning toward other solution: allowing user to move Guake to other monitor manually.

Easiest way to do this is to enable "Decorated" option in GTK app. Check this picture:

<img width="1920" height="1200" alt="Image" src="https://github.com/user-attachments/assets/f1f5aacc-12d8-4be0-aac2-583c79f05e2d" />

You can see that Guake title appears , and there is option to Move to Monitor (Left|Right|Up|Down).

@Davidy22  does not like that option , since "it  jars with how it's supposed to be a panel that pulls down from the top"

Anyway, if setting `decorated: true` to show Guake title is not an option, I plan to add the same option under "Preferences" to resolve this

**Comment by anhtumai** [comment]
for people using GNome, in order to deal with this in current codebase, you can use shortcut to manually move Guake drop-down terminal to others monitors: https://discourse.gnome.org/t/gnome-shortcut-key-to-move-window-to-next-monitor/15260
Super(Window) + Shift + Arrow key shortcut works for me.
